dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/hibernate/HibernateIdentifiableObjectStore.java

```diff
@@ -790,11 +789,22 @@ public Date getLastUpdated()
 
         CriteriaBuilder builder = getCriteriaBuilder();
 
-        JpaQueryParameters<T> jpaQueryParameters = new JpaQueryParameters<T>()
-            .addPredicates( getSharingPredicates( builder ) )
-            .addPredicate( root -> root.get( "uid" ).in( uids ) );
+        List<List<String>> uidPartitions = Lists.partition( new ArrayList<>( uids ), 20000 );
```

Comment:
Have you tested this flow when uids have around 20000 items or more, meaning the sql generated with the in clause would be quite huge. Asking because I have seen an IO error before when the sql was too long, but I dont remember the exact count of uids appended (I think i it was 70,000 or so).

Reply:
yes, tested with more than 50K elements and it worked. However making it smaller shouldn't change much, I might reduce to 10K or even less.